indico/web/templates/forms/datetime_widget.html

```diff
@@ -32,31 +67,36 @@
                     }
                 }
             }).data('onSelectHandlers', [
-                requireTime,
+                initTime,
                 function() {
                     dateField.trigger('change');
                 }
-            ]);
+            ]).on('change', function(e) {
+                updateTimeRequired(e);
+                updateResetFieldAction();
+            }).on('keydown', function(e) {
+                e.preventDefault();
+                if (e.which == K.BACKSPACE) {
+                    clearDate();
+                }
+            });
 
-            function requireTime() {
-                timeField.prop('required', true);
-            }
+            timeField.on('change', function applyOnChangeHandlers() {
+                var handlers = $(this).data('onChangeHandlers');
+                for (var i = 0; i < handlers.length; i++) {
+                    handlers[i].apply(this, arguments);
+                }
+            }).data('onChangeHandlers', [updateDateRequired, updateResetFieldAction]);
```

Comment:
is this modified elsewhere? if not why not simply call them directly inside the event handler?